docs/docs/reference/custom-components/01-component-package.md

```diff
@@ -0,0 +1,135 @@
+---
+sidebar_position: 1
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Creating a component package
+
+Custom components can be defined in ROS 2 packages and used alongside the standard library of AICA components in the
+AICA application framework. Components can be implemented in C++ or Python. A component package can contain multiple
+components in either language.
+
+## Package infrastructure
+
+A minimal directory structure for an example package named `custom_component_package` with C++ and Python component
+implementations is shown below.
+
+```
+custom_component_package
+│
+├── component_descriptions
+│   ├── custom_component_package_cpp_component.json
+│   └── custom_component_package_py_component.json
+│
+├── custom_component_package
+│   └── py_component.py
+│
+├── include
+│   └── custom_component_package
+│       └── CppComponent.hpp
+│
+├── src
+│   └── CppComponent.cpp
+│
+├── CMakeLists.txt
+├── package.xml
+└── setup.cfg
+```
+
+The following sections describe the package contents in more detail.
+
+### package.xml
+
+The package manifest file defines the package metadata (name, version, description, maintainer and license) and package
+dependencies. An example including minimal dependencies is given below.
+
+```xml title="package.xml"
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+    <version>0.0.1</version>
+    <name>custom_component_package</name>
+    <description>An example package for custom components</description>
+    <maintainer email="john@example.com">John Doe</maintainer>
+    <license>MIT</license>
+
+    <buildtool_depend>ament_cmake_auto</buildtool_depend>
+    <buildtool_depend>ament_cmake_python</buildtool_depend>
+
+    <depend>modulo_components</depend>
+
+    <export>
+        <build_type>ament_cmake</build_type>
+    </export>
+</package>
+```
+
+### CMakeLists.txt
+
+The package build instructions are defined in a `CMakeLists.txt` file.
+
+```cmake title="CMakeLists.txt"
+cmake_minimum_required(VERSION 3.15)
+project(custom_component_package)
+
+# find dependencies
+find_package(ament_cmake_auto REQUIRED)
+find_package(ament_cmake_python REQUIRED)
+
+ament_auto_find_build_dependencies()
+
+include_directories(include)
+
+# highlight-start
+### Register and install C++ Components here ###
```

Comment:
Can you control the color of the highlight? Black font on grey background is a bit hard to read

Reply:
Yeah it looks good in dark mode but quite bad in light mode. I'll see if I can adjust it, since the ability to highlight a line is still quite nice.